handlers/validator.go

```diff
@@ -1785,7 +1802,12 @@ func ValidatorStatsTable(w http.ResponseWriter, r *http.Request) {
 		pubKey, err := hex.DecodeString(strings.Replace(vars["index"], "0x", "", -1))
 		if err != nil {
 			logger.Errorf("error parsing validator public key %v: %v", vars["index"], err)
-			http.Error(w, "Internal server error", http.StatusInternalServerError)
+			SetPageDataTitle(data, "Validator not found")
+			data := InitPageData(w, r, "validators", fmt.Sprintf("/validator/%v", vars["index"]), "", validatorNotFoundTemplateFiles)
+
+			if handleTemplateError(w, r, "validator.go", "Validator", "GetValidatorDeposits", validatorNotFoundTemplate.ExecuteTemplate(w, "layout", data)) != nil {
+				return // an error has occurred and was processed
+			}
 			return
 		}
 		index, err = db.GetValidatorIndex(pubKey)
```

Comment:
If we get an error here, it still shows the error page, you can try with `0x1234`